### PR TITLE
[CodeGen] Update ROCm Intrinsic Rule

### PR DESCRIPTION
**Pull request by petrex**
The PR updates ROCm intrinsic rule.

- dispatch through OCML instead of LLVM pure implementation

- add few math routines

### CODE REVIEW

File: src/codegen/llvm/intrin_rule_rocm.cc [hunk 1]
```diff
@@ -45,27 +44,28 @@ inline void DispatchExternOCML(const TVMArgs& args, TVMRetValue* rv) {
...
-.set_body(DispatchLLVMPureIntrin<::llvm::Intrinsic::round, 1>);
+.set_body(DispatchExternOCML);
 
 TVM_REGISTER_GLOBAL("tvm.intrin.rule.rocm.trunc")
-.set_body(DispatchLLVMPureIntrin<::llvm::Intrinsic::trunc, 1>);
+.set_body(DispatchExternOCML);
 
 TVM_REGISTER_GLOBAL("tvm.intrin.rule.rocm.fabs")
-.set_body(DispatchLLVMPureIntrin<::llvm::Intrinsic::fabs, 1>);
+.set_body(DispatchExternOCML);
```
Comment (masahi) [inline src/codegen/llvm/intrin_rule_rocm.cc]: What is the motivation for this change? Is it faster?

Comment (petrex) [reply src/codegen/llvm/intrin_rule_rocm.cc]: @masahi Thanks for the feedback.

Yes, OCMLs are intermediate representations of low-level mathematical functions.

The benefits would be 

- Fixed, simple API so it scales across different architectures.  There are experts working on optimization SW implementations for newer architecture. We might get performance gain with later SW/HW release.

Comment (petrex) [reply src/codegen/llvm/intrin_rule_rocm.cc]: btw if you are asking fabs, in particular, I would say the perf would be probably quite similar. We get perf gain in other math routines.
